// to update: python -m grpc_tools.protoc -I./ --python_out=. --pyi_out=. --grpc_python_out=. ./backup.proto
syntax = "proto3";

package backup;

// work with google drive through this service
service GoogleDrive {
    rpc Backup (DBRequest) returns (BackupReply);
    rpc DownloadLatestBackup (Empty) returns (DownloadReply);
    rpc Healthcheck (DBRequest) returns (HealthcheckReply);
}

message DBRequest {
    string db_host = 1;
    uint32 db_port = 2;
    string db_username = 3;
    string db_password = 4;
    string db_name = 5;
}

message BackupReply {
    string file_id = 1;
}

message DownloadReply {
    bytes file_content = 1;
}

message HealthcheckReply {
    bool is_ok = 1;
}

message Empty {
}
